### DISCUSSION TITLE
possible extension of current offerings for templates into a system for third party addons

### DISCUSSION THREAD

**Discussion by SilkieSabra** [OpenCollarTeam/OpenCollar#989] [Ideas]
Phidoux, with coding and testing help mostly from Yosty and Ping,  has been working on a template for addons that would do considerably more than just serve as the link code between an external device and oc_addons menu.  
This concept would basically create a simplified version of the oc core, to be dropped into other devices.  These devices could then connect to AddOns, or be usable without wearing the collar (analogous to how the AO now works).  There are currently three scripts in development:  oc_standalone_core, oc_standalone_states and oc_standalone_dialog.  
I see some advantageous in producing such a system.  First, it would allow users to easily adapt a lot of older devices out there to work with addons and avoid triggering the Highlander code (which is what we call the code that generates a forcible detach for a second device with opencollar code).  Second, it might solve our problem with generating multiple leashes and leashpoints.
I have three questions I would like to work toward consensus on, assuming all bugs can be worked out.
1) **Permissions**.  Should the system be GPL2 or public domain? Our other templates are public domain to enable third party creators to set permissions as they choose on their own work without invoking the open source license.
2) **Mission Support**.  Is this a bridge too far for OpenCollar?  At the very least we would have to impose a limit for ourselves on supporting the work of third party developers using this code.
3) **What to Call It**.  I admit i'm not fond of "stand alone" because it's a contradiction with "addon".  It's vague and confusing.  We've batted around renaming the existing oc_plugin_template to oc_apps_template to match oc_addons_template.  Then renaming this oc_attachment-system, oc_hybrid_system, oc_peripheral_system?  Any thoughts?

**Comment by Omnia2021** [comment]
oc_chimera_system. It describes it but it is a bit heavy maybe, just a thought  :)
If OpenCollar wants to stay current we have to take leaps forward. I also think staying open source is very important to the integrity of the brand. Just a few thoughts  :)

**Comment by SilkieSabra** [comment]
Public Domain:  This means anyone can use this script for any purpose, it is not licensed.  We have a few of these but they are limited only to link code so that third party creators can link their own devices to an OpenCollar menu without having to set their device to open source.

**Comment by tayaphidoux** [comment]
i have been thinking about it and i think capsule might work better than standalone oc_capsule_addon since its a wrapper for oc_plugin to work as an addon/standalone
